### DISCUSSION TITLE
An inexpensive hardware solution for those that are struggling with configuring software-based remotes

### DISCUSSION THREAD

**Discussion by DVoorhees1957** [GerryDazoo/Slinger#398] [General]
**NOTE: The Broadlink RM4 is not the only device that provides the functionality described in this post; it just happens to be the one that I have personally tested. People replying to this thread have mentioned using other devices, such as the MOES UFO-R6. I can't personally vouch for anything other than the Broadlink RM4, so please be sure to do your own research before investing in a hardware remote solution such as this.**

I know a lot of good work has already been done on providing customized images of specific remotes, mapping the buttons to specific key codes, etc... But for anyone that's struggling with Slingbox remote BIN files, and/or getting such a remote working, I wanted to highlight a solution that is completely independent of either your Slingbox, or Slinger. It is called the BroadLink RM4 Mini IR Universal Remote Control, and I recently purchased one on Amazon for about $21; the link is https://smile.amazon.com/gp/product/B07ZSF46BX/. (There's also a slightly more expensive version that works with both IR and RF remotes.)

Basically, it's a very small IR learning remote (1.5 inches tall and 2 inches in diameter) that you put in the same room as the device (or devices) you want to control. In my case, I have a Sony TV and 3 TiVos, and each TiVo has a different unit address (so that remote commands can be sent to just one TiVo without affecting the others). It comes with a library of official and user-supplied remotes, but I think it's strength is that it is also a learning remote; you can essentially create your own custom remote, label the buttons as you wish, and have the Broadlink RM4 learn the code to be associated with each button. It took me about 15 minutes to define 25 different buttons for my first TiVo, and about half that time to define and learn the buttons for my 2nd TiVo.

You communicate with the Broadlink RM4 from anywhere in the world via an iPhone app, and the lag between the time you press a button and seeing the effect on the device in question isn't that bad (something like a few seconds). I'm very happy with the acquisition, and it lets me not have to open a web browser to send remote control key codes to the Slingbox via Slinger. In fact, it frees me from having to worry at all about what remote BIN file is currently loaded in my Slingbox!

Anyway, it's an inexpensive and very effective solution, for anyone that's interested.

**Comment by Toei79** [comment]
you already try it? the key its the remote connection , i never see some like that kinda cool.

i have logitech harmony.

**Reply by DVoorhees1957** [reply]
Yes, I’ve already set it up (with 2 of my 3 TiVos, because I only have 2 Slingboxes) and it works really well.

**Comment by mdehrlich** [comment]
I have FIOS cable, and I found the MOES UFO-R6 a better solution.  It does have many predefined remotes, and has a learning mode where it can be trained.  Also works with alexa and google to let you use voice commands to control some things.  Can control your tv, amp and cable box.

**Comment by Pthepeg** [comment]
This is a great solution. Thanks for sharing. It seems to me this would also let me change/upgrade my set top box or go to a different provider and still use my slingbox.

**Comment by tmcafee52** [comment]
Is this still the best solution for 350's that were added after the November 9 cutoff?  I bought 2 350's on eBay and they work fine with the RM4 Mini.  I was wondering if there was any success in getting the HTML remotes to work.  I have to switch between screens to control the TV using the RM4 interface but the web page approach has all of the controls on a single page.

**Reply by mdehrlich** [reply]
I use an HTML remote for Verizon  FIOS.  I have two slingboxes and I use 3 configs on each with differing resolutions and bandwidth settings.  My remote lets me switch between them without having to enter the URL for each remote config.  
https://github.com/GerryDazoo/Slinger/discussions/446#discussioncomment-4481448

**Reply by tmcafee52** [reply]
I haven't been able to change the definition for the remote in the slingbox because the setup hangs with the "initializing" screen.  Is there something I can do to define the proper remote in the 350?

**Reply by GerryDazoo** [reply]
The Sling Player Desktop needs access to the Sling Servers to work, so depending on how you may have blocked access for your slingboxes it could be affecting where you are running SPD.
